test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -220,16 +231,108 @@ std::string ProtoConverter::fixedByteValueAsString(unsigned _width, unsigned _co
 	//      <-----------total length --------->
 	// Note: This assumes that maskUnsignedIntToHex() invokes toHex(..., HexPrefix::Add)
 	unsigned startPos = 66 - numMaskNibbles;
+	// Extracts the least significant numMaskNibbles from the result
+	// of maskUnsignedIntToHex().
+	return maskUnsignedIntToHex(
+		_counter,
+		numMaskNibbles
+	).substr(startPos, numMaskNibbles);
+}
+
+std::string ProtoConverter::hexValueAsString(
+	unsigned _numBytes,
+	unsigned _counter,
+	bool _isHexLiteral,
+	bool _decorate
+)
+{
+	solAssert(_numBytes <= 32, "Proto ABIv2 fuzzer: Invalid hex length");
+
+	// If _decorate is set, then we return a hex"" or a "" string.
+	if (_numBytes == 0)
+		return Whiskers(R"(<?decorate><?isHex>hex</isHex>""</decorate>)")
+			("decorate", _decorate)
+			("isHex", _isHexLiteral)
+			.render();
 
-	// Extracts the least significant numMaskNibbles from the result of "maskUnsignedIntToHex",
-	// and replaces "0x" with "hex\"...\"" string.
 	// This is needed because solidity interprets a 20-byte 0x prefixed hex literal as an address
 	// payable type.
-	return Whiskers(R"(hex"<value>")")
-		("value", maskUnsignedIntToHex(_counter, numMaskNibbles).substr(startPos, numMaskNibbles))
+	return Whiskers(R"(<?decorate><?isHex>hex</isHex>"</decorate><value><?decorate>"</decorate>)")
+		("decorate", _decorate)
+		("isHex", _isHexLiteral)
+		("value", croppedString(_numBytes, _counter, _isHexLiteral))
 		.render();
 }
 
+std::string ProtoConverter::variableLengthValueAsString(
+	unsigned _numBytes,
+	unsigned _counter,
+	bool _isHexLiteral
+)
+{
+	solAssert(_numBytes >= 0, "Proto ABIv2 fuzzer: Invalid hex length");
+	if (_numBytes == 0)
+		return Whiskers(R"(<?isHex>hex</isHex>"")")
+			("isHex", _isHexLiteral)
+			.render();
+
+	// Since we create a larger literal by duplicating the first
+	// 32 bytes, we can cache it. This saves us expensive calls
+	// to keccak256.
+	unsigned batchSize = _numBytes >= 32 ? 32 : _numBytes;
+	std::string cachedString = hexValueAsString(
+		batchSize,
+		_counter,
+		_isHexLiteral,
+		/*decorate=*/false
+	);
+	_numBytes -= batchSize;
+	std::string output = cachedString;
+	// Check if size of cached string meets expectations
+	bool sanityCheck = false;
+	if (_numBytes > 0)
+		sanityCheck = _isHexLiteral ?
+			cachedString.size() == 64 :
+			cachedString.size() == 32;
+
+	solAssert(
+		_numBytes == 0 || sanityCheck,
+		"Proto ABIv2 fuzzer: Logic flaw in variable literal creation"
+	);
+
+	// Append cached literal until we are able to produce an
+	// output string that is _numBytes in size.
+	while (_numBytes > 0)
```

Comment:
Other than the initial assignment to `output`, it's only used here again. Isn't it? So maybe add a function call `output = concatOutput(output, cachedString, numBytes ...)`